test/extract-file-paths.test.ts

```diff
@@ -0,0 +1,42 @@
+import { Stats } from 'fs';
+import * as findModule from '../lib/find';
+import { MAX_SUPPORTED_FILE_SIZE } from '../lib/common';
+
+describe('extract file paths', () => {
+  it('should skip files greater than 2GB', async () => {
+    const statSpy = jest.spyOn(findModule, 'stat');
+
+    const statWithAllowedFileSize = {
+      isFile: () => true,
+      isDirectory: () => false,
+      size: MAX_SUPPORTED_FILE_SIZE - 1,
+    } as Stats;
+
+    const statWithFileSizeGreaterThanMaxAllowed = {
+      isFile: () => true,
+      isDirectory: () => false,
+      size: MAX_SUPPORTED_FILE_SIZE + 1,
+    } as Stats;
+
+    statSpy.mockResolvedValueOnce(statWithAllowedFileSize);
+    statSpy.mockResolvedValueOnce(statWithFileSizeGreaterThanMaxAllowed);
+    statSpy.mockResolvedValueOnce(statWithAllowedFileSize);
+
+    const absolutePathWithAllowedFileSize =
+      'fake-absolute-path-with-allowed-file-size';
+    const absolutePathWithNotAllowedFileSized =
+      'fake-absolute-path-with-not-allowed-file-size';
```

Comment:
This was only a draft suggestion, I think instead of 'fake-absolute...' we can go with 

const absolutePathWithAllowedFileSize = path.join('non-existing-path', 'add.cpp');
const absolutePathWithAllowedFileSize = path.join(''non-existing-path', 'main.cpp');

Similar way to make it more real and closer to what you raised here

https://github.com/snyk/snyk-cpp-plugin/pull/49/commits/0926b19e153a90b68ce947ae0e633ceb7c7056b6#diff-93d1526c040533e0c2144cc266765205b6ea97c38bbac8584611f9430709b497R37

do want us to use real file paths as we will not introduce a 2GB fixture here 😄

Reply:
Yea usually I want to use a fake path as well, for similar things. I can make it a new path.

In the other PR I used a real path, because it's easier and cleaner to get it to work than to get the spy to mock the readdir function (readdir has been overriden by multiple versions, so it collides).